test/integration/pages/desktop/download_page.js

```diff
@@ -23,5 +29,41 @@ class DownloadPage extends Page {
   download() {
     return browser.click(this.downloadButton);
   }
+
+  setUseCase(id, useCase) {
```

Comment:
Could you explain your thought process behind this? I am having a hard time seeing exactly what it does.

Reply:
I thought that we would have an object which keeps track of all the uploaded files by the tests so other tests can also use them in future. So I created `uploadedFilesUseCase` private object which stores `{ 'archive-#id': 'some_use_case' } `  key-value pairs. That's why we have `setUseCase` and `getUploadedFiles` methods.

For example let's say one test uploaded two files called `file1.txt` and `file2.txt` for testing if uploads work and set's it's USE_CASE to `UPLOAD_TESTING` and we then ran a couple of tests in between so `file1.txt` and `file2.txt` were no longer the topmost files in DOM then there was another test case which checked if downloading works fine, so instead of uploading new files again it can get the two files used for upload testing by calling `getUploadedFiles('UPLOAD_TESTING')` and then it can locate the two files by querying DOM with the archive-id's returned by the `getUploadedFiles` function.

I took some inspiration from the [link that you mentioned](https://github.com/mozilla/notes/blob/master/test/integration/page_objects/list_page.js#L50) and created `UploadedFile` class.

But now I see that this creates a relationship between tests ( one test depends on another - change in one requires change in another too - which is bad ) 

Now I think as @dannycoates mentioned this was an overkill.

Reply:
> Yes, I read #810 before commenting. I understand the problem, it's this solution that I don't. The `useCase` concept is confusing and doesn't seem necessary, `UploadedFile` shouldn't be in `download_page.js` (probably `home_page.js`), and `getUploadedFiles` should probably get populated from a selector.
> 
> The idea is fine, just the implementation needs work. A better solution will probably present itself once the code is in the right place, `home_page.js`.

Please let me know if I should start working in this direction.

Reply:
You have the right idea but an object isn't needed. My link was just the object to represent the items on the page. If you see [this](https://github.com/mozilla/notes/blob/master/test/integration/page_objects/list_page.js#L26) line, you will see that we create a new list of objects and return it each time that function is called. 

This is how we can get around having a separate object that must keep track of the elements. Each time we call that function it will get and create a list of objects representing what is on the page at that specific moment. 

Let me know if that helps.